### PR TITLE
[TIR] Make loop unrolling in LoopPartition optional

### PR DESCRIPTION
**Pull request by mbaret**
For certain analysis/tensorization, it can be useful to keep the loop structure when partitioning loops. The current behaviour removes For loops of length 1. This change introduces the option to preserve these loops with the 'unroll' flag.

### CODE REVIEW

**Comment by mbaret** [comment]
cc @tqchen

**Review by ANSHUMAN87 — COMMENTED** [review]
Please find some high level comments.
Could you please share if it is possible, in which scenario you need to preserve the single iter loops ?

**Comment by ANSHUMAN87** [comment]
Also cc @MarisaKirisame !

File: src/tir/transforms/loop_partition.cc [hunk 1]
```diff
@@ -40,9 +40,13 @@ namespace tir {
 
 struct LoopPartitionConfigNode : public tvm::AttrsNode<LoopPartitionConfigNode> {
   bool partition_const_loop;
+  bool no_unroll_single_iter_loop;
```
Comment (tqchen) [inline src/tir/transforms/loop_partition.cc:43]: how about `unroll_loop_with_extent_one` and default to true?

**Comment by mbaret** [comment]
In terms of why I need this, I want to do some pattern matching on loops to determine if they correspond to a particular operation. This is made a lot more consistent if loops of extent 1 aren't dropped from the IR.

**Comment by tqchen** [comment]
Thanks @mbaret I understand where you are coming from. Ideally we would also like to develop tools to be able to detect patterns that also removes these trivial loops. But we can tackle that later

**Comment by ANSHUMAN87** [comment]
> In terms of why I need this, I want to do some pattern matching on loops to determine if they correspond to a particular operation. This is made a lot more consistent if loops of extent 1 aren't dropped from the IR.

Thanks @mbaret , i got your point and concur in case of pattern matching. Also I totally support the idea of having more control over optimization techniques :)

**Review by ANSHUMAN87 — APPROVED** [review]
LGTM provided CI is totally green this time :slightly_smiling_face:

**Review by giuseros — APPROVED** [review]
Thank for addressing my comments! LGTM

**Review by tqchen — APPROVED** [review]

**Pull request merged**